distribution/errors.go

```diff
@@ -84,7 +84,9 @@ func continueOnError(err error) bool {
 func retryOnError(err error) error {
 	switch v := err.(type) {
 	case errcode.Errors:
-		return retryOnError(v[0])
+		if len(v) != 0 {
+			return retryOnError(v[0])
+		}
```

Comment:
right, we would probably remember that we should change the test in https://github.com/docker/docker/pull/21212#discussion_r56320243